scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -728,6 +782,51 @@ def _thread(handler, timeout, outdir, logfile, fifo_fn, pid_fn, results, harness
         in_fp = open(fifo_out, "rb", buffering=0)
         log_out_fp = open(logfile, "wt")
 
+        return out_fp, in_fp, log_out_fp
+
+    @staticmethod
+    def _thread_close_files(fifo_in, fifo_out, pid, out_fp, in_fp, log_out_fp):
+        log_out_fp.close()
+        out_fp.close()
+        in_fp.close()
+        if pid:
+            try:
+                if pid:
+                    os.kill(pid, signal.SIGTERM)
+            except ProcessLookupError:
+                # Oh well, as long as it's dead! User probably sent Ctrl-C
+                pass
+
+        os.unlink(fifo_in)
+        os.unlink(fifo_out)
+
+    @staticmethod
+    def _thread_update_instance_info(handler, handler_time, out_state):
+        handler.instance.execution_time = handler_time
+        if out_state == "timeout":
+            handler.instance.status = "failed"
+            handler.instance.reason = "Timeout"
+        elif out_state == "failed":
+            handler.instance.status = "failed"
+            handler.instance.reason = "Failed"
+        elif out_state in ['unexpected eof', 'unexpected byte']:
+            handler.instance.status = "failed"
+            handler.instance.reason = out_state
+        else:
+            handler.instance.status = out_state
+            handler.instance.reason = "Unknown"
+
+    @staticmethod
+    def _thread(handler, timeout, outdir, logfile, fifo_fn, pid_fn, results,
```

Comment:
Maybe as a next step could be good to extract methods related to Thread to a separate Class (instead of using `@staticmethod` and calling extracted method with `QEMUHandler._thread...`. You can check that class:
https://github.com/zephyrproject-rtos/zephyr/blob/2e333000b5c4d316224bf850449d947824a77951/scripts/pylib/pytest-twister-harness/src/twister_harness/device/fifo_handler.py